src/roadveh.h

```diff
@@ -83,6 +83,8 @@ void RoadVehUpdateCache(RoadVehicle *v, bool same_length = false);
 void GetRoadVehSpriteSize(EngineID engine, uint &width, uint &height, int &xoffs, int &yoffs, EngineImageType image_type);
 
 struct RoadVehPathCache {
+	Trackdir origin_td = INVALID_TRACKDIR;
+	TileIndex origin_tile = INVALID_TILE;
```

Comment:
The RoadVehPathCache structure is saved in the savegame, and these two new members need to be stored as well.
You need to add them to the saveload data and add a SL-version bump.

Reply:
You're right. Desync confirmed. Save is required indeed.
https://i.imgur.com/gUmDN4J.png